### PR TITLE
Inherit .hound.yml from org-level repo

### PR DESCRIPTION
**Pull request by gylaz**
If org-level configuration is enabled, then all enabled repos that fall
under that org should respect and use its `.hound.yml` file.

### CODE REVIEW

**Comment by gylaz** [comment]
[Org-level doesn't work without `.hound.yml` in the repo -- but should](https://trello.com/c/W7sx7U9Y/803-org-level-doesnt-work-without-houndyml-in-the-repo-but-should)

File: app/models/hound_config.rb [hunk 1]
```diff
@@ -19,17 +20,17 @@ class HoundConfig
     Linter::Swift => { default: true },
     Linter::Tslint => { default: false },
   }.freeze
-  CONFIG_FILE = ".hound.yml"
 
-  attr_reader_initialize :commit
+  attr_initialize [:commit!, :owner!]
```
Comment (salbertson) [inline app/models/hound_config.rb]: I'm still not used to these helpers.

File: app/models/owner.rb [hunk 2]
```diff
@@ -2,21 +2,13 @@ class Owner < ApplicationRecord
...
   rescue ActiveRecord::RecordNotUnique => exception
-    Raven.capture_exception(
-      exception,
-      extra: {
-        github_id: github_id,
-        name: name,
-      },
-    )
-
+    capture_exception(exception, name, github_id)
```
Comment (salbertson) [inline app/models/owner.rb]: `Raven.capture_exception` was distracting, or what?

File: spec/models/hound_config_spec.rb [hunk 3]
```diff
@@ -39,101 +38,133 @@
...
+        commit = stub_commit(".hound.yml" => repo_hound_config)
+        owner = owner_stub(owner_hound_config)
+        hound_config = build_hound_config(commit, owner)
+
+        expect(hound_config).to be_enabled_for("flake8")
+      end
+    end
+
+    context "when linter enabled at owner-level but disabled at repo-level" do
+      it "returns true" do
```
Comment (salbertson) [inline spec/models/hound_config_spec.rb]: Returns true or false?

File: spec/models/hound_config_spec.rb [hunk 4]
```diff
@@ -39,101 +38,133 @@
...
+        repo_hound_config = ""
+        commit = stub_commit(".hound.yml" => repo_hound_config)
+        owner = owner_stub(owner_hound_config)
+        hound_config = build_hound_config(commit, owner)
+
+        expect(hound_config).to be_enabled_for("flake8")
+      end
+    end
+
+    context "when linter enabled at owner-level but disabled at repo-level" do
```
Comment (salbertson) [inline spec/models/hound_config_spec.rb]: `when linter is enabled...`?

File: app/models/hound_config.rb (hunk 1 above)
Comment (salbertson) [reply app/models/hound_config.rb]: What do you think about default argument values of `EmptyCommit.new` and `MissingOwner.new`?

File: app/models/owner.rb (hunk 2 above)
Comment (gylaz) [reply app/models/owner.rb]: Yeah, big block of unimportant code...

File: app/models/hound_config.rb (hunk 1 above)
Comment (gylaz) [reply app/models/hound_config.rb]: I don't think the default arguments would do us any good. Most of the time, we want the caller to pass to use something other than `EmptyCommit`. Just calling `HoundConfig.new` isn't really useful. WDYT?

File: spec/models/hound_config_spec.rb (hunk 4 above)
Comment (gylaz) [reply spec/models/hound_config_spec.rb]: I did this consciously, since adding `is` pushes the line over 80 chars.

**Pull request merged**